datamanager/splitLargeZip/split_zip.py

```diff
@@ -0,0 +1,112 @@
+        with zipfile.ZipFile(path, "r") as zip_handle:
+            zip_handle.extractall(temp_zip_contents)
+            top_level_paths = {p.name: p for p in Path(temp_zip_contents).joinpath(path.stem).iterdir()}
+
+            Configs.schema_path = top_level_paths["schema.json"]
+            Configs.start_size += Configs.schema_path.stat().st_size
+
+            Configs.split_path = top_level_paths["split.csv"]
+            Configs.start_size += Configs.split_path.stat().st_size
+
+            split_files(top_level_paths["images"], top_level_paths["latest"], temp_zip_contents)
+
+
+def split_files(images_path: Path, latest_path: Path, folder_contents_path: str):
+    image_names = list()
+    current_size = Configs.start_size
+    archive_number = 1
+    dict_image_paths = {path.name: path for path in list(Path(images_path).iterdir())}
+    dict_latest_paths = {path.stem: path for path in list(Path(latest_path).iterdir())}
+
+    for image_name in dict_image_paths:
+        size = dict_latest_paths[image_name].stat().st_size + dict_latest_paths[image_name].stat().st_size
+        current_size += size
+        if current_size < ONE_GB:
+            image_names.append(image_name)
+        else:
+            write_one_zip(folder_contents_path, image_names, archive_number)
+            archive_number += 1
+            current_size = Configs.start_size + size
+            image_names = list()
+            image_names.append(image_name)
+    write_one_zip(folder_contents_path, image_names, archive_number)  # for the last zip
+
+
+def write_one_zip(folder_contents_path: str, image_names: list, archive_number: int):
+    with zipfile.ZipFile("splitArchive" + str(archive_number) + ".zip", "w") as zip_handle:
+        for root, _, files in os.walk(folder_contents_path):
+            for file in files:
+                for image_name in image_names:
+                    if image_name in file:
+                        zip_handle.write(
+                            Path(root).joinpath(file),
+                            Path(root).joinpath(file).relative_to(folder_contents_path),
+                        )
+        zip_handle.write(
+            Path(root).joinpath(Configs.schema_path),
+            Path(root).joinpath(Configs.schema_path).relative_to(folder_contents_path),
+        )
+        processed_split_name = process_split_file_one_archive(image_names)
+        zip_handle.write(
+            Path(root).joinpath(processed_split_name),
+            Path(root).joinpath(Configs.split_path).relative_to(folder_contents_path),
+        )
+
+
+def process_split_file_one_archive(images: list):
+    lines = list()
+
+    with open(Configs.split_path, "rb") as f:
+        reader = csv.reader(TextIOWrapper(f, "utf-8"), delimiter="\t")
+        lines.append(next(reader))
+        for row in reader:
+            if row[0] in images:
+                lines.append(row)
+
+    temp_split_path = str(Configs.split_path) + "_"
```

Comment:
Use the `suffix` param from `create_archive` to generate the `temp_split_path`, that way we can debug splits easier